### PR TITLE
Prepare for v2 support

### PR DESCRIPTION
**Pull request by Rauno56**
The oldest current version supported is from 8 months ago. I'd like to support at least v2(2.0.0 released in 2017, latest 2.x in 2021). However, the initialization is a bit different for that. This PR prepares a few of the things without actually adding support for v2 or changing any of the behaviour to make the next PR more concise and easier to follow.

Check the commits for details, but:

* supports initialization in the way v2 does it - version checking is a bit hacky TBH, I'm open to a better way if you have any preferences;
* catches async errors in tests(stumbled upon those implementing v2 support);
* renames events "ping" and "pong" in tests since those are internally used for heartbeats in v2.

Look out for any linting/formatting inconsistencies, because I didn't find a way to automatically apply those(I see GHA job for prettier but running default prettier gave me tons of markdown changes as well - just isolate to `ts` files?).

### CODE REVIEW

**Comment by blumamir** [comment]
Thanks @Rauno56 ,
@mottibec  wrote this instrumentation, can you please take a look?

Regarding the prettier, I usually run `yarn prettier` from the root of the monorepo and it works.
Can you share how exactly you run prettier in your setup?

**Comment by Rauno56** [comment]
Problem solved with linting! Thanks, @blumamir!

**Comment by blumamir** [comment]
> Problem solved with linting! Thanks, @blumamir!

Great :)
we used to have husky hook running the prettier automatically on git push, but unfortunately husky doesn't work well with monorepos:
https://github.com/typicode/husky/issues/677

I see now in the issue that it might be fixed in version 6 of husky, which we can test and verify one day

**Comment by blumamir** [comment]
> > Problem solved with linting! Thanks, @blumamir!
> 
> Great :)
> we used to have husky hook running the prettier automatically on git push, but unfortunately husky doesn't work well with monorepos:
> [typicode/husky#677](https://github.com/typicode/husky/issues/677)
> 
> I see now in the issue that it might be fixed in version 6 of husky, which we can test and verify one day

opened an issue for that: https://github.com/aspecto-io/opentelemetry-ext-js/issues/152

File: packages/instrumentation-socket.io/test/utils.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    const server = http.createServer();
+    const sio = createServerInstance(server);
+    server.listen(0, () => {
+        const port = (server.address() as AddressInfo).port;
+        callback(sio, port);
+    });
+};
+
+export const createServerInstance = (server?: http.Server) => {
+    if (version && /^2\./.test(version)) {
```
Comment (blumamir) [inline packages/instrumentation-socket.io/test/utils.ts:29]: as current support is only for version 2, we are expecting this `if` to always be truthy, right?

**Review by blumamir — APPROVED** [review]

**Comment by blumamir** [comment]
LGTM
@mottibec - do you want to take a look as well?

File: packages/instrumentation-socket.io/test/utils.ts (hunk 1 above)
Comment (Rauno56) [reply packages/instrumentation-socket.io/test/utils.ts:29]: Current support is for v3 and v4, so the opposite - we currently expect it to be always falsy.

**Comment by mottibec** [comment]
LGTM
Thanks @Rauno56

File: packages/instrumentation-socket.io/test/utils.ts (hunk 1 above)
Comment (blumamir) [reply packages/instrumentation-socket.io/test/utils.ts:29]: Right :)

**Pull request merged**